### DISCUSSION TITLE
Not possible to easily extract offending edge/node ID from VueFlowError

### DISCUSSION THREAD

**Discussion by snoozbuster** [bcakmakoglu/vue-flow#1277] [Ideas]
### Current Behavior

When a VueFlowError [is instantiated](https://github.com/bcakmakoglu/vue-flow/blob/4b4bcdccbb75733950005f613c9460fc35769fab/packages/core/src/utils/store.ts#L15), a reference to the offending edge ID is passed. This is used to generate [an error message](https://github.com/bcakmakoglu/vue-flow/blob/master/packages/core/src/utils/errors.ts#L45) but the value itself is not stored on the error, either in a custom property or in the `cause` property of the error. This makes string parsing the only way to extract the offending edge/node/etc, instead of simply accessing it off the error object, which makes error handling of the type mentioned in [this comment](https://github.com/bcakmakoglu/vue-flow/pull/1149#issuecomment-1911679338) quite challenging/error-prone.

### Expected Behavior

The error should expose all relevant information for programmatic access.

### Steps To Reproduce

1. Trigger any EDGE_INVALID error and observe the resulting error.

**Comment by bcakmakoglu** [comment]
Not a bug but a feature request, so I'm moving it to discussions.

**Comment by bcakmakoglu** [comment]
Would something like this work for you?

```ts
import { ErrorCode, isErrorOfType } from '@vue-flow/core'

const { onError } = useVueFlow()

onError((err) => {
	// typeguard to assert the type of the args provided by the error - if you're not using TS, this step is not necessary and would just be the same as doing `error.code === ErrorCode.EDGE_INVALID`
	if (isErrorOfType(err, ErrorCode.EDGE_INVALID) {
		const [id] = err.args

		console.log(id) // 'e1-2'
	}
})

**Reply by bcakmakoglu** [reply]
I'll release it as proposed above in `1.31.0` :)

**Reply by snoozbuster** [reply]
yes, this would work fine. thanks!
